include/electionguard/precompute_buffers.hpp

```diff
@@ -41,13 +41,15 @@ namespace electionguard
         Triple &operator=(const Triple &triple);
         Triple &operator=(Triple &&);
 
-        std::unique_ptr<ElementModQ> get_exp() { return exp->clone(); }
-
-        std::unique_ptr<ElementModP> get_g_to_exp() { return g_to_exp->clone(); }
+        std::unique_ptr<Triple> clone();
 
-        std::unique_ptr<ElementModP> get_pubkey_to_exp() { return pubkey_to_exp->clone(); }
+        std::unique_ptr<ElementModQ> clone_exp() { return exp->clone(); }
+        std::unique_ptr<ElementModP> clone_g_to_exp() { return g_to_exp->clone(); }
+        std::unique_ptr<ElementModP> clone_pubkey_to_exp() { return pubkey_to_exp->clone(); }
 
-        std::unique_ptr<Triple> clone();
+        ElementModQ *get_exp() const { return exp.get(); }
```

Comment:
this is a minor change to the existing logic behavior. Previously `get` really meant `clone` in this context, which meant we were _always_ copying the values even if we didn't need to. Now we have `get` and `clone` which return a pointer and a unique pointer respectively. That made it possible to remove about 60-70% of the copies that were happening when making encryptions and disjunctive selection proofs. might have a minor performance benefit but will definitely impact the memory footprint positively. It also allowed me to adjust some of the apis to accept a `const TwoTriplesAndAQuad&` instead of `unique_ptr<TwoTriplesAndAQuad>` which is more explicit for indicating when a method is just accessing a variable versus taking ownership of it.